tests/cli-v2.rs

```diff
@@ -1374,3 +1374,53 @@ fn check_pgp_keys() {
         );
     })
 }
+
+#[test]
+fn install_allow_downgrade() {
+    clitools::setup(Scenario::MissingComponent, &|config| {
+        let trip = TargetTriple::from_build();
+
+        // this dist has no rls and there is no newer one
+        set_current_dist_date(config, "2019-09-14");
+        expect_ok(
+            config,
+            &[
+                "rustup",
+                "toolchain",
+                "install",
+                "nightly",
+                "--no-self-update",
+            ],
+        );
+        expect_err(
+            config,
+            &[
+                "rustup",
+                "toolchain",
+                "install",
+                "nightly",
+                "--no-self-update",
+                "-c",
+                "rls",
+            ],
+            &format!(
+                "component 'rls' for target '{}' is unavailable for download for channel nightly",
+                trip,
+            ),
+        );
```

Comment:
Perhaps here add a check that the `rls` component is not installed  You can do that with:

```rust
expect_component_not_executable(config, "rls");
```

Reply:
Although I added it I would rather see that in a different testcase which checks if toolchain install does not change anything without allow-downgrade. When I look at the subcommand toolchain install I see 7 arguments and would expect at least as many as 7 tests at the level of cli-v2.rs, but there are not. Or else I don't know yet about all the test levels. Well that's obvious :)